include/deal.II/numerics/vector_tools.templates.h

```diff
@@ -884,6 +888,178 @@ namespace VectorTools
   }
 
 
+  template <int dim, typename VectorType, int spacedim, int fe_degree>
+  void project_parallel (const Mapping<dim, spacedim>   &mapping,
+                         const DoFHandler<dim,spacedim> &dof,
+                         const ConstraintMatrix         &constraints,
+                         const Quadrature<dim>          &quadrature,
+                         const std::function< typename VectorType::value_type (const typename DoFHandler<dim, spacedim>::active_cell_iterator &, const unsigned int)> func,
+                         VectorType                     &vec_result)
+  {
+    const parallel::Triangulation<dim,spacedim> *parallel_tria =
+      dynamic_cast<const parallel::Triangulation<dim,spacedim>*>(&dof.get_tria());
+    Assert (parallel_tria !=0, ExcNotImplemented());
+
+    const MPI_Comm &mpi_communicator = parallel_tria->get_communicator();
+
+    typedef typename VectorType::value_type number;
```

Comment:
Can you move this line after the `Asserts` (just above L 915). I first missed it because it is in the same block as the `Assert` and I was very confused how the code was compiling.